hudi-utilities/src/main/java/org/apache/hudi/utilities/deltastreamer/DeltaSync.java

```diff
@@ -162,18 +162,23 @@ public DeltaSync(HoodieDeltaStreamer.Config cfg, SparkSession sparkSession, Sche
     this.fs = fs;
     this.onInitializingHoodieWriteClient = onInitializingHoodieWriteClient;
     this.props = props;
-    this.schemaProvider = schemaProvider;
 
     refreshTimeline();
-
     this.transformer = UtilHelpers.createTransformer(cfg.transformerClassNames);
     this.keyGenerator = DataSourceUtils.createKeyGenerator(props);
-
-    this.formatAdapter = new SourceFormatAdapter(
-        UtilHelpers.createSource(cfg.sourceClassName, props, jssc, sparkSession, schemaProvider));
-
     this.conf = conf;
+    refreshSchemaProvider(schemaProvider);
+  }
 
+  /**
+   * Very useful when DeltaStreamer is running in continuous mode.
+   * @param schemaProvider
+   * @throws IOException
+   */
+  public void refreshSchemaProvider(SchemaProvider schemaProvider) throws IOException {
```

Comment:
@pratyakshsharma : It looks like refreshSchemaProvider  not only refreshes schema-provider but also recreates Source and setup WriteClient

@vinothchandar : Recreating DeltaSync each run would require to handle embedded timeline server reuse and async compaction triggering differently.  Another option is to have explicit refreshSchema() API in SchemaProvider (with default implementation (for compatibility) and implementing refresh in existing Schema Provider implementation) and have delta-streamer call this ? Let me know your thoughts on this ?

Reply:
right.. embedded server re-use is important. let's scratch my idea.

Reply:
> It looks like refreshSchemaProvider not only refreshes schema-provider but also recreates Source and setup WriteClient

Do you see any side effects of doing this? @bvaradar 

> have delta-streamer call this ? 

This call will happen exactly at the same point where I am calling refreshSchemaProvider in delta-streamer, right?